legion/engine/physics/mesh_splitter_utils/mesh_half_edge.hpp

```diff
@@ -0,0 +1,268 @@
+#pragma once
+#include <core/core.hpp>
+#include <physics/mesh_splitter_utils/splittable_polygon.h>
+#include <physics/physics_statics.hpp>
+
+namespace legion::physics
+{
+    struct MeshHalfEdge : std::enable_shared_from_this<MeshHalfEdge>
+    {
+        math::vec3 position;
+        math::vec2 uv;
+
+        std::shared_ptr<MeshHalfEdge> nextEdge = nullptr;
+        std::shared_ptr<MeshHalfEdge> pairingEdge = nullptr;
+        std::shared_ptr<MeshHalfEdge> shadowEdge = nullptr;
+
+        std::weak_ptr<SplittablePolygon> owner;
+
+        //TODO: these bools should probably be stored as an array in SplittablePolygon
```

Comment:
perhaps this can be done?

Reply:
this is optimization related, not a priority at the moment since we are splitting simple meshes